### PR TITLE
v2 beta!

### PR DESCRIPTION
**Pull request by sblunt**
Won't release v2 yet, but this PR will be the bulk of it. Changes include:

- Fitting in new bases!
- compute_all_orbits separated out
- Hip IAD!
- param_idx now used everywhere under the hood (system parsing updated)
- rebound backend!
(... anything I'm forgetting?)

### CODE REVIEW

File: orbitize/sampler.py [hunk 1]
```diff
@@ -62,6 +64,9 @@ def _logl(self, params):
         """
         # compute the model based on system params
         # jitter output from compute model
+
+        # TODO SOFIA: pass in self.system.rebound keyword here
```
Comment (semaphoreP) [inline orbitize/sampler.py:68]: remove?

File: orbitize/sampler.py [hunk 2]
```diff
@@ -294,7 +322,7 @@ def reject(self, samples):
                 has size ``num_samples``. This should be the output of \
                 ``prepare_samples()``.
 
-        Return:
+        Return:spectrum_index = trueorder - config['ordmin']
```
Comment (semaphoreP) [inline orbitize/sampler.py:325]: is this a typo?

File: orbitize/system.py [hunk 3]
```diff
@@ -144,245 +122,395 @@ def __init__(self, num_secondary_bodies, data_table, stellar_mass,
...
 
-        if len(self.rv[0]) > 0 and self.fit_secondary_mass:
-            # Rob and Lea:
-            # for instrument in rv_instruments:
-                # add gamma and sigma for each and label each unique gamma and sigma per instrument name (gamma+instrument1, ...)
-            for instrument in self.rv_instruments:
-                self.sys_priors.append(priors.UniformPrior(-5, 5))  # gamma prior in km/s
-                self.labels.append('gamma_{}'.format(instrument))
+        # Obtain extra necessary data to assign priors for XYZ
+        if fitting_basis == 'XYZ':
```
Comment (semaphoreP) [inline orbitize/system.py:143]: We should move this into the xyz class in the future. Can basically pass in the data table to the XYZ class to help it create priors.

File: orbitize/system.py [hunk 4]
```diff
@@ -1,6 +1,5 @@
 import numpy as np
-from orbitize import priors, read_input, kepler
-
+from orbitize import nbody, kepler, conversions, basis
```
Comment (semaphoreP) [inline orbitize/system.py:2]: is conversions being used at all?

Comment (sblunt) [reply orbitize/system.py]: nope, removed

File: orbitize/system.py (hunk 3 above)
Comment (sblunt) [reply orbitize/system.py:143]: good call. made an issue. Issue #271 .

File: orbitize/sampler.py (hunk 2 above)
Comment (sblunt) [reply orbitize/sampler.py]: yup, fixed

File: orbitize/sampler.py (hunk 1 above)
Comment (sblunt) [reply orbitize/sampler.py]: yup, done. thanks!

**Pull request merged**